build.sbt

```diff
@@ -318,28 +321,90 @@ lazy val cli = (project in file("examples") / "cli")
+    Compile / fullClasspath := (Compile / fullClasspath).value ++ (client / Compile / fullClasspath).value,
+    assembly / assemblyJarName := s"${name.value}_${scalaBinaryVersion.value}-${version.value}.jar",
+     */
+    assembly / logLevel := Level.Debug,
+    assembly / test := {},
+    assembly / assemblyShadeRules := Seq(
+      ShadeRule.rename("com.fasterxml.**" -> "shaded.@0").inAll,
+      ShadeRule.rename("org.antlr.**" -> "shaded.@0").inAll,
+    ),
+    assemblyPackageScala / assembleArtifact := false,
+
+    assembly / assemblyMergeStrategy := {
+      case PathList("META-INF", xs@_*) => MergeStrategy.discard
+      case _ => MergeStrategy.first
+    },
+    assembly / fullClasspath := {
+      val compileClasspath = (server / Compile / fullClasspath).value ++ (client / Compile / fullClasspath).value
+      val testClasses = (server / Test / products).value
+      compileClasspath ++ testClasses.map(Attributed.blank)
+    }
+  )
+
+val sparkVersion = "3.5.1"
 lazy val spark = (project in file("connectors/spark"))
-  .dependsOn(client % "compile->compile;test->test")
-  .dependsOn(server % "test->test")
   .settings(
     name := s"$artifactNamePrefix-spark",
-    scalaVersion := "2.13.14",
+    scalaVersion := "2.12.6",
     commonSettings,
     javaOptions ++= Seq(
       "--add-opens=java.base/sun.nio.ch=ALL-UNNAMED",
     ),
-    javaCheckstyleSettings(file("dev") / "checkstyle-config.xml"),
+    javaCheckstyleSettings(file("dev/checkstyle-config.xml")),
+    libraryDependencies ++= Seq(
+      "org.apache.spark" %% "spark-sql" % sparkVersion,
+      "com.fasterxml.jackson.core" % "jackson-databind" % "2.12.5",
+      "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.12.5",
+      "com.fasterxml.jackson.core" % "jackson-annotations" % "2.12.5",
+      "com.fasterxml.jackson.core" % "jackson-core" % "2.12.5",
+      "org.antlr" % "antlr4-runtime" % "4.9.3",
+      "org.antlr" % "antlr4" % "4.9.3",
+    ).map(_.excludeAll(
+      ExclusionRule("com.fasterxml.jackson.core", "jackson-databind"),
+      ExclusionRule("com.fasterxml.jackson.module", "jackson-module-scala"),
+      ExclusionRule("com.fasterxml.jackson.core", "jackson-annotations"),
+      ExclusionRule("com.fasterxml.jackson.core", "jackson-core"),
+      ExclusionRule("org.antlr", "antlr4-runtime"),
+      ExclusionRule("org.antlr", "antlr4")
+    )),
     libraryDependencies ++= Seq(
-      "org.apache.spark" %% "spark-sql" % "4.0.0-preview1",
       // Test dependencies
       "org.junit.jupiter" % "junit-jupiter" % "5.10.3" % Test,
+      "org.mockito" % "mockito-core" % "5.11.0" % Test,
+      "org.mockito" % "mockito-inline" % "5.2.0" % Test,
+      "org.mockito" % "mockito-junit-jupiter" % "5.12.0" % Test,
       "net.aichler" % "jupiter-interface" % JupiterKeys.jupiterVersion.value % Test,
-      "io.delta" %% "delta-spark" % "4.0.0rc1" % Test,
+      "org.apache.spark" %% "spark-core" % sparkVersion % Test classifier "tests",
```

Comment:
do we need this line? We already depend on spark-sql

Reply:
@ravivj-db good question .. does it work without this line?

Reply:
It's working without it.